lib/galaxy/tool_util/linters/stdio.py

```diff
@@ -1,29 +1,27 @@
 """This module contains a linting functions for tool error detection."""
+import re
+
+from ._util import node_props
 from .command import get_command
 
 
 def lint_stdio(tool_source, lint_ctx):
     tool_xml = getattr(tool_source, "xml_tree", None)
-    # determine line to report for general problems with stdio
-    try:
-        tool_line = tool_xml.find("./tool").sourceline
-        tool_path = tool_xml.getpath(tool_xml.find("./tool"))
-    except AttributeError:
-        tool_line = 0
-        tool_path = None
+    # determine node to report for general problems with stdio
+    tool_node = tool_xml.find("./tool")
```

Comment:
We're allowing `tool_xml` to be `None` here, so I guess you want to guard this with a `if tool_xml is not None:` ?

Reply:
I would suggest to move the linter tests for yaml and cwl to a separate PR. 

The problem here is that the main lint function should distinguish which linters to call based on the tool language and not the tool type. Thus, all the xml linters are called for the yaml tools.

Also with the suggested change the tool_type is still `None` (because the yaml file does not include it) and the